tests/test_visitors/test_ast/test_builtins/test_assign/test_unpacking_rules.py

```diff
@@ -13,15 +13,29 @@
 single_assignment = '{0} = 1'
 
 tuple_assignment1 = 'first, {0} = (1, 2)'
-tuple_assignment1 = '{0}, second = (1, 2)'
+tuple_assignment2 = '{0}, second = (1, 2)'
 
 list_assignment = '[first, second] = (1, 2)'
 
+nested_list_assignment0 = '[first, second], third = ((1, 2), 3)'
+nested_list_assignment1 = 'first, [second, third] = (1, (2, 3))'
+nested_list_assignment2 = 'first, (second, [third, fourth]) = (1, (2, (3, 4)))'
+
+multiple_level_nested_list_assignment = (
+    'first, [second, [third, fourth]] = (1, (2, (3, 4)))'
+)
+
 spread_assignment1 = '{0}, *second = [1, 2, 3]'
 spread_assignment2 = 'first, *{0} = [1, 2, 3]'
 
 spread_assignment_in_list = '[first, *rest] = [1, 2, 3]'
 
+nested_spread_assignment_in_list = 'first, [second, *rest] = [1, (2, 3, 4)]'
+
+multiple_level_nested_spread_assign_in_list = (
```

Comment:
One more case to test: `z = (a, [b, c]) = (1, (2, 3))`

Reply:
I have added support for such cases and much more in new commits.
I am happy that unpacking iterables is not supported when using walrus operator :smile: